types/v1/json.go

```diff
@@ -86,20 +106,32 @@ func convertJSONSpan(zs ZipkinJSONSpan) *Span {
 	return s
 }
 
-type Annotation struct {
-	Timestamp int64     `json:"timestamp"`
-	Value     string    `json:"value"`
-	Host      *Endpoint `json:"endpoint,omitempty"`
-}
-
-type binaryAnnotation struct {
-	Key      string    `json:"key"`
-	Value    string    `json:"value"`
-	Endpoint *Endpoint `json:"endpoint,omitempty"`
-}
+// guessAnnotationType takes a value and, if it is a string, turns it into a bool,
+// int64 or float64 value when possible. This is a workaround for the fact that
+// Zipkin v1 BinaryAnnotation values are always transmitted as strings.
+// (See e.g. the Zipkin API spec here:
+// https://github.com/openzipkin/zipkin-api/blob/72280f3/zipkin-api.yaml#L235-L245)
+//
+// However it considers the possibility that the value is not a string in case the
+// BinaryAnnotation does not implement the Zipkin API v1 spec. In this case it
+// will just return the same value, without modifying it. See this issue
+// for such an example:
+// https://github.com/honeycombio/honeycomb-opentracing-proxy/issues/37
+func guessAnnotationType(v interface{}) interface{} {
```

Comment:
This has some changes from https://github.com/honeycombio/honeycomb-opentracing-proxy/pull/41
which you may not want to include, but seems useful
only used by zipkin v1 api, since v2 takes any value type anyway

Reply:
I'd like to keep this function as it was in order to restrict this change to what is required to get the `v2` endpoint working. Let's consider this change in a separate PR.

Reply:
Agreed - I am taking a look at that change separately